packages/babel-core/src/transformation/block-hoist-plugin.js

```diff
@@ -41,24 +39,13 @@ const blockHoistPlugin = {
   visitor: {
     Block: {
       exit({ node }) {
-        let hasChange = false;
-        for (let i = 0; i < node.body.length; i++) {
-          const bodyNode = node.body[i];
-          if (bodyNode?._blockHoist != null) {
-            hasChange = true;
-            break;
-          }
-        }
-        if (!hasChange) return;
-
-        node.body = sortBy(node.body, function (bodyNode) {
-          let priority = bodyNode?._blockHoist;
-          if (priority == null) priority = 1;
-          if (priority === true) priority = 2;
-
-          // Higher priorities should move toward the top.
-          return -1 * priority;
-        });
+        if (node.body.every(node => node?._blockHoist == null)) return;
+        const priority = node => {
+          if (node?._blockHoist == null) return 1;
+          if (node?._blockHoist === true) return 2;
```

Comment:
```suggestion
          if (node?._blockHoist == null) return -1;
          if (node?._blockHoist === true) return -2;
```

This is potentially what caused the test failures, previous code set `priority` then returned the negated value.

Reply:
Ah, good catch :)  Thanks!